platforms/core-runtime/logging/src/main/java/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandler.java

```diff
@@ -94,7 +95,7 @@ public ProblemBuilder apply(ProblemBuilderDefiningLabel builder) {
                         ProblemBuilderDefiningLocation problemBuilderDefiningLocation = builder.label(usage.formattedMessage())
                             .documentedAt(usage.getDocumentationUrl());
                         return addPossibleLocation(diagnostics, problemBuilderDefiningLocation)
-                            .category(DEPRECATION, "generic_deprecation")
+                            .category(DEPRECATION, TextUtil.screamingSnakeToKebabCase(usage.getType().name()))// TODO (donat) we can further categorize deprecation warnings
```

Comment:
❓ Is there a ticket for this? Otherwise this TODO will stay here forever I'm afraid.